client/my-sites/importer/site-importer/site-importer-input-pane.jsx

```diff
@@ -0,0 +1,243 @@
+					importData: {
+						title: resp.site_title,
+						supported: resp.supported_content,
+						unsupported: resp.unsupported_content,
+						favicon: resp.favicon,
+						engine: resp.engine,
+					},
+					loading: false,
+					importSiteURL: siteURL,
+				} );
+			} )
+			.catch( err => {
+				this.setState( {
+					loading: false,
+					error: true,
+					errorMessage: `${ err.message }`,
+				} );
+			} );
+	};
+
+	importSite = () => {
+		this.setState( { loading: true }, this.resetErrors );
+
+		wpcom.wpcom.req
+			.post( {
+				path: `/sites/${ this.props.site.ID }/site-importer/import-site`,
+				apiNamespace: 'wpcom/v2',
+				formData: [
+					[ 'import_status', JSON.stringify( toApi( this.props.importerStatus ) ) ],
+					[ 'site_url', this.state.importSiteURL ],
+				],
+			} )
+			.then( resp => {
+				this.setState( { loading: false } );
+
+				const data = fromApi( resp );
+				const action = finishUpload( this.props.importerStatus.importerId )( data );
+				defer( () => {
+					Dispatcher.handleViewAction( action );
+				} );
+			} )
+			.catch( err => {
+				this.setState( {
+					loading: false,
+					error: true,
+					errorMessage: `${ err.message } (${ err.code })`,
+				} );
+			} );
+	};
+
+	resetImport = () => {
+		this.setState(
+			{
+				loading: false,
+				importStage: 'idle',
+			},
+			this.resetErrors
+		);
+	};
+
+	render() {
+		return (
+			<div className="site-importer__site-importer-pane">
+				{ this.state.importStage === 'idle' && (
+					<div>
+						<p>{ this.props.description }</p>
+						<div className="site-importer__site-importer-url-input">
+							<TextInput
+								disabled={ this.state.loading }
+								onChange={ this.setUrl }
+								value={ this.state.siteURLInput }
+							/>
+							<Button
+								disabled={ this.state.loading }
+								isPrimary={ true }
+								onClick={ this.validateSite }
+							>
+								{ this.props.translate( 'Continue' ) }
+							</Button>
+						</div>
+					</div>
+				) }
+				{ this.state.importStage === 'importable' && (
+					<div className="site-importer__site-importer-confirm-site-pane">
+						<SiteImporterSitePreview
+							siteURL={ this.state.importSiteURL }
+							importData={ this.state.importData }
+							isLoading={ this.state.loading }
+						/>
+						<div className="site-importer__site-importer-confirm-site-pane-container">
+							<Button disabled={ this.state.loading } onClick={ this.importSite }>
+								{ this.props.translate( 'Yes! Start import' ) }
+							</Button>
+							<Button
+								disabled={ this.state.loading }
+								isPrimary={ false }
+								onClick={ this.resetImport }
+							>
+								{ this.props.translate( 'No' ) }
+							</Button>
+						</div>
+					</div>
+				) }
+				{ this.state.error && (
+					<ErrorPane type="importError" description={ this.state.errorMessage } />
+				) }
+			</div>
+		);
+	}
+}
+
+const mapDispatchToProps = dispatch => ( {
+	mapAuthorFor: importerId => ( source, target ) =>
```

Comment:
I may be wrong, but would it not be worth creating a new action for this? It seems like a lot of logic for the `mapDispatchToProps ` cb

Reply:
It was the easiest way to achieve the functionality. I agree it would make more sense to extract is a separate action, but with the underlying Flux code, I don't think it makes sense to add more things there instead of focusing on migrating to Redux and improving the code here after that.

Reply:
That makes sense :)